aiidalab_widgets_base/structures.py

```diff
@@ -4,126 +4,113 @@
 import tempfile
 import datetime
 from collections import OrderedDict
-from traitlets import Bool
+import numpy as np
 import ipywidgets as ipw
+from traitlets import Instance, Unicode, Union, link, observe, validate
 
-from aiida.orm import CalcFunctionNode, CalcJobNode, Node, QueryBuilder, WorkChainNode, StructureData
+# ASE imports
+from ase import Atom, Atoms
+from ase.data import chemical_symbols
+
+# AiiDA and AiiDA lab imports
+from aiida.orm import CalcFunctionNode, CalcJobNode, Data, QueryBuilder, Node, StructureData, WorkChainNode
+from aiida.plugins import DataFactory
 from .utils import get_ase_from_file
 
 
-class StructureManagerWidget(ipw.VBox):  # pylint: disable=too-many-instance-attributes
+class StructureManagerWidget(ipw.VBox):
     '''Upload a structure and store it in AiiDA database.
 
-    Useful class members:
-    :ivar has_structure: whether the widget contains a structure
-    :vartype has_structure: bool
-    :ivar frozen: whenter the widget is frozen (can't be modified) or not
-    :vartype frozen: bool
-    :ivar structure_node: link to AiiDA structure object
-    :vartype structure_node: StructureData or CifData'''
+    Attributes:
+        structure(Atoms): trait that contains the selected structure. 'None' if no structure is selected.
+        structure_node(StructureData, CifData): trait that contains AiiDA structure object
+        node_class(str): trait that contains structure_node type (as string).
+    '''
+
+    structure = Union([Instance(Atoms), Instance(Data)], allow_none=True)
+    structure_node = Instance(Data, allow_none=True, read_only=True)
+    node_class = Unicode()
 
-    has_structure = Bool(False)
-    frozen = Bool(False)
-    DATA_FORMATS = ('StructureData', 'CifData')
+    DATA_FORMATS = {'CifData': 'cif', 'StructureData': 'structure'}
```

Comment:
Should this be `SUPPORTED_DATA_FORMATS`?

Reply:
agreed.